propolis/src/block.rs

```diff
@@ -108,67 +181,113 @@ impl<R: BlockReq> PlainBdev<R> {
             }
 
             if let Some(mut req) = reqs.pop_front() {
-                let res = match req.oper() {
-                    BlockOp::Flush => {
-                        let res =
-                            unsafe { libc::fdatasync(self.fp.as_raw_fd()) };
-                        if res == -1 {
-                            BlockResult::Failure
-                        } else {
-                            BlockResult::Success
-                        }
-                    }
-                    BlockOp::Read => self.process_read(&mut req, ctx),
-                    BlockOp::Write => self.process_write(&mut req, ctx),
-                };
-                req.complete(res, ctx);
+                let result = self.process_request(&mut req, ctx);
+                req.complete(result, ctx);
             } else {
                 reqs = self.cond.wait(reqs).unwrap();
             }
         }
     }
-    fn process_read(&self, req: &mut R, ctx: &DispCtx) -> BlockResult {
+
+    // Match against individual BlockReqs and call appropriate processing functions.
+    fn process_request(&self, req: &mut R, ctx: &DispCtx) -> BlockResult {
         let mem = ctx.mctx.memctx();
 
         let mut offset = req.offset();
+
+        // TODO: are buffers guaranteed to be sector size? do BlockDevBackingStores require
```

Comment:
I'm pretty sure buffers are not required to be sector size.

Reply:
Removed comment in cf484fc